serverjs/analytics.js

```diff
@@ -1,112 +1,65 @@
-/* used in the now abandoned GetTokens attempt to get token counts from rules text. If  
-var Small = {
-  'zero': 0,
-  'one': 1,
-  'two': 2,
-  'three': 3,
-  'four': 4,
-  'five': 5,
-  'six': 6,
-  'seven': 7,
-  'eight': 8,
-  'nine': 9,
-  'ten': 10,
-  'eleven': 11,
-  'twelve': 12,
-  'thirteen': 13,
-  'fourteen': 14,
-  'fifteen': 15,
-  'sixteen': 16,
-  'seventeen': 17,
-  'eighteen': 18,
-  'nineteen': 19,
-  'twenty': 20,
-};
-
-var a, n, g;
-
-function text2num(a) {
-  return Small[a];  
-}
-*/
-function CheckContentsEqualityOfArray(target, candidate) {
-  var isValid = candidate.length == target.length;
-  if (!isValid) return false;
-
-  for (idx = 0; idx < target.length; idx++) {
-    if (!candidate.includes(target[idx])) {
-      isValid = false;
-      break;
-    }
-  }
-  return isValid;
-}
-
 function GetColorCat(type, colors) {
   if (type.toLowerCase().includes('land')) {
     return 'l';
-  } else if (colors.length == 0) {
+  }
+  if (colors.length === 0) {
```

Comment:
I'm not a fan of removing the else's here.

Reply:
Then I would not use a `return`... if `type.toLowerCase().includes('land')` the rest of the function does not execute because of the explicit `return 'l';` making the else redundant.

Reply:
we can also so something like:

```
let colorCat;

if (<something>) {
  ...
} else if (<another thing>) {
  ...
}
return colorCat;
```

Reply:
I like removing elses when not necessary